src/Fixer/ClassNotation/VisibilityRequiredFixer.php

```diff
@@ -138,11 +150,13 @@ protected function applyFix(\SplFileInfo $file, Tokens $tokens): void
                 continue;
             }
 
-            if (null !== $staticIndex) {
-                if ($this->isKeywordPlacedProperly($tokens, $staticIndex, $index)) {
-                    $index = $staticIndex;
+            $swapIndex = $staticIndex ?? $readOnlyIndex;
+
+            if (null !== $swapIndex) {
+                if ($this->isKeywordPlacedProperly($tokens, $swapIndex, $index)) {
+                    $index = $swapIndex;
                 } else {
-                    $this->moveTokenAndEnsureSingleSpaceFollows($tokens, $staticIndex, $index);
+                    $this->moveTokenAndEnsureSingleSpaceFollows($tokens, $swapIndex, $index);
```

Comment:
needs more testing with `readonly static` combinations

Reply:
no other action needed `Fatal error: Static property Foo::$prop1 cannot be readonly in tmp/test.php on line 4`